src/app/shared/code-editor-list/code-editor-list.service.ts

```diff
@@ -0,0 +1,291 @@
+import { Injectable } from '@angular/core';
+import { DescriptorLanguageService } from '../entry/descriptor-language.service';
+import { SourceFile, ToolDescriptor, Validation } from '../swagger';
+import { FileCategory } from './code-editor-list.component';
+
+/**
+ * This service currently only handles CWL, WDL, NFL, and Galaxy. All plugin languages and SERVICE will not work correctly.
+ *
+ * @export
+ * @class CodeEditorListService
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class CodeEditorListService {
+  static readonly NEXTFLOW_CONFIG_PATH = '/nextflow.config';
+  static readonly NEXTFLOW_PATH = '/main.nf';
+  constructor() {}
+  /**
+   * Determines whether to show the current sourcefile based on the descriptor type and tab
+   *
+   * @static
+   * @param {(SourceFile.TypeEnum | null | undefined)} sourcefileType
+   * @param {FileCategory} fileCategory
+   * @param {ToolDescriptor.TypeEnum} descriptorType
+   * @returns {boolean} Whether to show sourcefile or not
+   * @memberof CodeEditorListService
+   */
+  static showSourcefile(
```

Comment:
Just curious, because I've wondered about which way to tackle this before and I'm not clear on when to choose which, what is your reasoning for having static methods instead of injecting an instance of the service into the constructor of classes using this?

Reply:
If it can be made static, make it static. This file will likely change in the future because I'll be trying to merge the descriptor language bean from the backend with the frontend which means I'll need to inject a few things into this service.